client/state/reader/lists/actions.ts

```diff
@@ -0,0 +1,369 @@
+ * @param lists - Lists received
+ * @returns Action object
+ */
+export function receiveLists( lists: ReaderList[] ): ReaderListAction {
+	return {
+		type: READER_LISTS_RECEIVE,
+		lists,
+	};
+}
+
+/**
+ * Request the current user's subscribed lists.
+ * @returns Action object
+ */
+export function requestSubscribedLists(): ReaderListAction {
+	return {
+		type: READER_LISTS_REQUEST,
+	};
+}
+
+export function createReaderList( list: ReaderList ): ReaderListAction {
+	return { type: READER_LIST_CREATE, list };
+}
+
+/**
+ * Request a single Reader list.
+ * @param listOwner - List owner
+ * @param listSlug - List slug
+ * @returns Action object
+ */
+export function requestList( listOwner: string, listSlug: string ): ReaderListAction {
+	return { type: READER_LIST_REQUEST, listOwner, listSlug };
+}
+
+/**
+ * Receive a single Reader list.
+ * @param data - List data
+ * @param data.list - Reader list object
+ * @returns Action object
+ */
+export function receiveReaderList( data: { list: ReaderList } ): ReaderListAction {
+	return { type: READER_LIST_RECEIVE, data };
+}
+
+export function handleRequestListFailure( errorInfo: ErrorInfo ): ReaderListAction {
+	return {
+		type: READER_LIST_REQUEST_FAILURE,
+		error: errorInfo.error,
+		owner: errorInfo.owner,
+		slug: errorInfo.slug,
+	};
+}
+
+export function receiveCreateReaderList( data: { list: ReaderList } ): ReaderListAction {
+	return {
+		type: READER_LIST_CREATE_SUCCESS,
+		data,
+	};
+}
+
+export function handleCreateReaderListFailure( errorInfo: ErrorInfo ): ReaderListAction {
+	return {
+		type: READER_LIST_CREATE_FAILURE,
+		error: errorInfo.error,
+		owner: errorInfo.owner,
+		slug: errorInfo.slug,
+	};
+}
+
+/**
+ * Follow a list.
+ * @param listOwner - List owner
+ * @param listSlug - List slug
+ * @returns Action object
+ */
+export function followList( listOwner: string, listSlug: string ): ReaderListAction {
+	return {
+		type: READER_LIST_FOLLOW,
+		listOwner,
+		listSlug,
+	};
+}
+
+/**
+ * Receive a successful list follow.
+ * @param list - Followed list
+ * @returns Action object
+ */
+export function receiveFollowList( list: ReaderList ): ReaderListAction {
+	return {
+		type: READER_LIST_FOLLOW_RECEIVE,
+		list,
+	};
+}
+
+/**
+ * Unfollow a list.
+ * @param listOwner - List owner
+ * @param listSlug - List slug
+ * @returns Action object
+ */
+export function unfollowList( listOwner: string, listSlug: string ): ReaderListAction {
+	return {
+		type: READER_LIST_UNFOLLOW,
+		listOwner,
+		listSlug,
+	};
+}
+
+/**
+ * Receive a successful list unfollow.
+ * @param list - Unfollowed list
+ * @returns Action object
+ */
+export function receiveUnfollowList( list: ReaderList ): ReaderListAction {
+	return {
+		type: READER_LIST_UNFOLLOW_RECEIVE,
+		list,
+	};
+}
+
+/**
+ * Triggers a network request to update a list's details.
+ * @param list - List details to save
+ * @returns Action object
+ */
+export function updateReaderList( list: ReaderList ): ReaderListAction {
```

Comment:
It looks like the component/file that is currently using this is not using TypeScript. Should we leave the null validation for this one still?